crates/driver/src/infra/blockchain/contracts.rs

```diff
@@ -46,6 +57,10 @@ impl Contracts {
     pub fn weth_address(&self) -> eth::WethAddress {
         self.weth.address().into()
     }
+
+    pub fn ethflow_address(&self) -> Option<eth::ContractAddress> {
+        self.ethflow
+    }
```

Comment:
This wouldn't be needed with hooks support right? Ideally, we would keep this out of the `driver`.

Reply:
Although the ethflow contract holds enough `ETH` for all the trades it may not hold enough `WETH` for all the trades. So unless I'm missing something this is needed unfortunately.
I guess one thing to get around this is to use the balance fetching implementation that considers pre-interactions for all orders that need them and not updating the user's regular token balances when including orders because we don't know whether the pre-interaction would actually use some of the user's balance.

Reply:
Hmm, I'm not sure I understand. Ethflow orders use pre-interactions that wrap `WETH`. So if we simulate the `WETH` balance after executing the order's pre-interaction, we will get sufficient `WETH` to trade all user's Ethflow orders (since it always wraps all of its ETH).

Reply:
The point I was trying to make was more general. Right now we: 
* collect all tuples `(owner, sell_token)`
* fetch these balances
* go through all orders and decrement the corresponding user balance until nothing is left

If we want to use the balance fetcher we have to simulate the balance of each order individually because grouping them doesn't really make sense because they could have vastly different pre-interactions which get the required `sell_token` balance.
That's why I suggested to keep doing what are currently doing for orders that don't use any pre-interactions and for all the other orders we could simply check if the user has the expected balance after simulating the pre-interaction.
But this could be problematic when a user has 10 WETH, 1 order selling 10 WETH and 1 order selling 10 WETH with some random pre-interaction.

The point I'm trying to make is that we have way less guarantees about sell token balances with orders that include pre-interactions.

Reply:
BTW I tried going down the road of making pre-interaction aware balance fetching available in the `driver` but I think I first have to implement the old `CodeSimulating` trait for the driver `infra::Simulator` and for that I think it would be beneficial to move `ethrpc` into its own crate first.